### PR TITLE
Feature/update column dropdown

### PR DESCRIPTION
**Pull request by davidebukali**
#449

### CODE REVIEW

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { Form, Checkbox, CheckboxProps, DropdownItemProps, Segment } from 'semantic-ui-react';
+
+interface SelectQueryBuilderProps {
+  options: DropdownItemProps[];
+  selectedOptions?: string[];
+  onUpdateOptions?: (options: string) => void;
+}
+
+const segment_style = {
```
Comment (edwinmp) [inline frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: Stick to the naming convention

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  const onChange = (_event: React.SyntheticEvent<HTMLElement, Event>, data: CheckboxProps) => {
+    let updatedCheckboxes: string[] | undefined = [];
+    if (data.checked) {
+      updatedCheckboxes = checkboxes?.concat(data.value as string);
+    } else {
+      updatedCheckboxes = checkboxes?.filter((checkbox) => {
+        return checkbox !== data.value;
+      });
+    }
```
Comment (edwinmp) [inline frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: Rewrite into a single statement

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    addCheckboxes(updatedCheckboxes);
+  };
+
+  const isChecked = (value: string) => {
+    if (props.selectedOptions && props.selectedOptions.length > 0) {
+      const item = props.selectedOptions.find((c: string) => c === value);
+
+      return item ? true : false;
```
Comment (edwinmp) [inline frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: Rewrite without the true or false.

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    } else {
+      return false;
+    }
```
Comment (edwinmp) [inline frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: You don't need the else if all you're doing is returning.

**Review by edwinmp — CHANGES_REQUESTED** [review]
Tried to paste a screenshot and failed ... anyway, rearrange the checkboxes into 3 shorter columns instead of one super long one.
One way to do it is to wrap each 3 items in a `div` with the class `row`, and give each item the class `col-md-4`.





Rearrange them into

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx (hunk 1 above)
Comment (davidebukali) [reply frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: Added 3 columns for large sources and 2 or 1 for smaller sources. Also updated naming

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx (hunk 2 above)
Comment (davidebukali) [reply frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: Updated

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx (hunk 3 above)
Comment (davidebukali) [reply frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: Also updated

File: frontend/src/components/CheckboxGroup/CheckboxGroup.tsx (hunk 4 above)
Comment (davidebukali) [reply frontend/src/components/CheckboxGroup/CheckboxGroup.tsx]: Removed redundant else

**Review by edwinmp — APPROVED** [review]

**Pull request merged**